tests/components/gotify/test_config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Test the gotify config flow."""
+from unittest.mock import patch
+
+from homeassistant import config_entries, setup
+from homeassistant.components.gotify.config_flow import CannotConnect, InvalidAuth
+from homeassistant.components.gotify.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import RESULT_TYPE_CREATE_ENTRY, RESULT_TYPE_FORM
+
+
+async def test_form(hass: HomeAssistant) -> None:
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == RESULT_TYPE_FORM
+    assert result["errors"] == {}
+
+    with patch("homeassistant.components.gotify.config_flow.validate_input"), patch(
+        "homeassistant.components.gotify.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {"host": "https://1.1.1.1", "token": "test-token", "name": "test"},
```

Comment:
Minor nit: 1.1.1.1 is a real public IP address. [`example.com`](https://www.iana.org/domains/reserved) is reserved for these purposes.